tests/functional/spec/journeys/hub_and_spoke/hub_and_spoke.spec.js

```diff
@@ -159,7 +159,7 @@ describe("Feature: Hub and Spoke", () => {
 
     it("When the user returns to the Hub and continues, Then they should progress to the next section", async () => {
       await click(EmploymentTypeBlockPage.submit());
-      await expect(browser).toHaveUrlContaining(HubPage.url());
+      await expect(browser).toHaveUrl(expect.stringContaining(HubPage.url()));
```

Comment:
Strange that just below this we use a different method of asserting the URL 🤔 e.g.
const expectedUrl = await browser.getUrl();
await expect(expectedUrl).toContain(ProxyPage.url());

Reply:
Also, I though this is an issue that predates this PR, I think we need to be more strict with the check for the hub page, because `/questionnaire` is in all the page titles

Reply:
Ah yeah that is done a bit differently 🤔  According the the official docs, using `.toHaveUrl()` includes waiting and retry (and seems to have more flexibility with params), so I would think that would be the preferable function to use? 

I think I'll update the other occurrences where we're using `await browser.getUrl()` while we're here in this PR

Reply:
> Also, though this is an issue that predates this PR, I think we need to be more strict with the check for the hub page, because /questionnaire is in all the page titles

As discussed externally, a new helper function for strict checking of the [URL path was created for this](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1535/commits/6e413d1ac5711bf20e5a60b562f76a0420244605#diff-5624047e3f252f3afd475bc7b98f4607ae6f53887752347aca54d19ea7640b12R389-R393). Thanks for your help 👍

Reply:
> I think I'll update the other occurrences where we're using await browser.getUrl() while we're here in this PR

Now updated with in [this commit](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1535/commits/e1ae742eaffc3e6aecb24e8b5d9c85e92cdc2363). I've also updated the PR description to cover these changes. Thanks 👍